### PR TITLE
[AIRFLOW-3410] Add feature to allow Host Key Change for SSH Op

### PR DESCRIPTION
**Pull request by kaxil**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. 
  - https://issues.apache.org/jira/browse/AIRFLOW-3410

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
This PR adds a feature to allow connecting to host (that already has an entry in Known Host file with different key) that has a new IP. Currently, it fails with `Remote Identification Changed`.

I have also added documentation on adding extras field to SSH Connection

image

image

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: docs/howto/manage-connections.rst [hunk 1]
```diff
@@ -350,3 +350,51 @@ Extra (optional)
...
+
+Extra (optional)
+    Specify the extra parameters (as json dictionary) that can be used in ssh
+    connection. The following parameters out of the standard python parameters
+    are supported:
+
+    * **timeout** - An optional timeout (in seconds) for the TCP connect.
+    * **compress** - ``true`` to ask the remote client/server to compress traffic; `false` to refuse compression.
+    * **no_host_key_check** - Set to ``false`` to restrict connecting to hosts with no entries in ``~/.ssh/known_hosts`` (Hosts file). This provides maximum protection against trojan horse attacks, but can be troublesome when the ``/etc/ssh/ssh_known_hosts`` file is poorly maintained or connections to new hosts are frequently made. This option forces the user to manually add all new hosts. Default is ``true``, ssh will automatically add new host keys to the user known hosts files
+    * **allow_host_key_change** - Set to ``true`` if you want to allow connecting to hosts that has host key changed or when you get 'REMOTE HOST IDENTIFICATION HAS CHANGED' error. Default is ``false``.
```
Comment (ashb) [inline docs/howto/manage-connections.rst]: Add a warning (here too) about not preventing MiTM attacks, and that the other work around is to remove the host from the `~/.ssh/known_hosts` file on the workers?

File: docs/howto/manage-connections.rst [hunk 2]
```diff
@@ -350,3 +350,51 @@ Extra (optional)
...
+
+    Example "extras" field:
+
+    .. code-block:: json
+
+       {
+          "timeout": "10",
+          "compress": "false",
+          "no_host_key_check": "false",
+          "allow_host_key_change": "true"
```
Comment (ashb) [inline docs/howto/manage-connections.rst]: People will copy this by default - so can the first example people see be secure please?

**Review by ashb — CHANGES_REQUESTED** [review]

File: docs/howto/manage-connections.rst (hunk 2 above)
Comment (kaxil) [reply docs/howto/manage-connections.rst]: Updated

**Comment by kaxil** [comment]
@ashb Updated :)

File: docs/howto/manage-connections.rst [hunk 3]
```diff
@@ -350,3 +350,51 @@ Extra (optional)
...
+    * **allow_host_key_change** - Set to ``true`` if you want to allow connecting to hosts that has host key changed or when you get 'REMOTE HOST IDENTIFICATION HAS CHANGED' error.  This wont protect against Man-In-The-Middle attacks. Other possible solution is to remove the host entry from ``~/.ssh/known_hosts`` file. Default is ``false``.
+
+    Example "extras" field:
+
+    .. code-block:: json
+
+       {
+          "timeout": "10",
+          "compress": "false",
+          "no_host_key_check": "true",
```
Comment (ashb) [inline docs/howto/manage-connections.rst]:
```suggestion
          "no_host_key_check": "false",
```

too please.

**Review by ashb — APPROVED** [review]
Make this change then LGTM. (Pre-appproving.)

**Pull request merged**